core/src/main/java/org/opentosca/toscana/core/data/Platform.java

```diff
@@ -0,0 +1,49 @@
+package org.opentosca.toscana.core.data;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+
+public class Platform {
+
+    public final String id;
+    public final String name;
+    public final Set<Property> requiredProperties;
+
+    /**
+     * Creates a new platform.
+     * @param id short identifier of platform. must match regex [a-z_-]+
+     * @param name displayable name of platform. must not be an empty string
+     * @param requiredProperties the properties the platform requires.
+     */
+    public Platform(String id, String name, Set<Property> requiredProperties){
+        assert id != null && id.matches("[a-z_-]+");
```

Comment:
As said in the LogEntry Class use Exceptions instead of assert.

Reply:
fixed